src/main/kotlin/com/autonomousapps/internal/utils/gradleStrings.kt

```diff
@@ -50,30 +50,25 @@ private fun ProjectComponentIdentifier.identityPath(): String {
 }
 
 internal fun ResolvedArtifactResult.toCoordinates(): Coordinates {
-  val gradleVariantIdentification = variant.toGradleVariantIdentification()
-  val resolved = id.componentIdentifier.toCoordinates(gradleVariantIdentification)
+  return variant.wrapInIncludedBuildCoordinates(id.componentIdentifier)
+}
+
+private fun ResolvedVariantResult.wrapInIncludedBuildCoordinates(id: ComponentIdentifier): Coordinates {
+  val variantIdentification = toGradleVariantIdentification()
+  val resolved = id.toCoordinates(variantIdentification)
 
   // Doesn't resolve to a project, so can't be an included build. Return as-is.
   if (resolved !is ProjectCoordinates) return resolved
 
-  // may be a composite substitution
-  val identity = ProjectCoordinates(
-    (id.componentIdentifier as ProjectComponentIdentifier).identityPath(),
-    gradleVariantIdentification
-  )
-
-  // Identity path matches project path, so we assume this isn't resolved from an included build, and return as-is.
-  if (resolved == identity) return resolved
```

Comment:
Not doing this now leads to all **ProjectCoordinates** being wrapped into **IncludedBuildCoordinates**. Overall, I think this is a good change to move a more consistent setup.

We could add some other check here. We could check, if we are in an included build setup at all. That is, for builds that do not include anything, we could still do the traditional approach which would keep this behavior for most builds.

However, to do that we would need more context passed into the method. At lease some `runningWithIncludedBuilds: Boolean` parameter. I thought about this, but it would ass more complexity without any additional functionality. It would just be a safeguard to avoid using the "new" approach in "traditional" builds.